src/ralph/admin/widgets.py

```diff
@@ -156,14 +161,63 @@ def get_related_url(self, info, action, *args):
 
     def value_from_datadict(self, data, files, name):
         value = data.get(name)
-        if getattr(self, 'multi', False) is True:
-            print('name', name)
+        if self.multi:
             if value:
-                value = value.split(',')
+                value = value.split(self.separator)
             else:
                 value = []
         return value
 
+    def get_search_fields(self):
+        searched_fields_tooltip = defaultdict(list)
+        polymorphic_descendants = getattr(
+            self.rel_to, '_polymorphic_descendants', []
+        )
+
+        if polymorphic_descendants:
+            # Check if model after which we are looking for is polymorphic
+            # if they are also looking for the models of its dependencies
+            # or by limit_models defined in model
+            polymorphic_models = polymorphic_descendants
+            limit_models = getattr(self.field, 'limit_models', [])
+            if limit_models:
+                polymorphic_models = [
+                    get_model(*i.split('.')) for i in limit_models
+                ]
+
+            for related_model in polymorphic_models:
+                for field_name in self.admin_site._registry[
+                    related_model
+                ].search_fields:
+                    field = get_field_by_relation_path(
+                        related_model, field_name
+                    )
+                    key = str(related_model._meta.verbose_name)
+                    searched_fields_tooltip[key].append(
+                        str(field.verbose_name)
+                    )
+        else:
+            for field_name in self.admin_site._registry[self.rel_to].search_fields:
+                try:
+                    field = get_field_by_relation_path(
```

Comment:
this block is the same as above - you could extract it to some function and call it here and above (to process search fields of single model)